presto-raptor/src/main/java/com/facebook/presto/raptor/storage/organization/ShardOrganizer.java

```diff
@@ -111,6 +133,12 @@ public int getShardsInProgress()
         return shardsInProgress.size();
     }
 
+    @Managed
+    public int getShardsDeltaInProgress()
+    {
+        return shardsInProgress.values().stream().filter(Optional::isPresent).collect(toSet()).size();
```

Comment:
Metric exposed to jmx should be light weighted as it can be call frequently by other service. If we can not get delta in progress in a easy way I suggest we remove this metric.